### ISSUE TITLE
[Feature] Execute process_batch_result in a separate process when enabling overlap scheduling

### ISSUE THREAD

**Issue by vipwangerxiao** [sgl-project/sglang#10514]
### Checklist

- [x] 1. If the issue you raised is not a feature but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [x] 2. Please use English, otherwise it will be closed.

### Motivation

We tested and found that after disabling the overlap scheduler, the Mean TTFT on H20 decreased by 32%, from 2464.08ms to 1676.25ms.

The main reason is that after enabling the overlap scheduler, in 

> resolve_last_batch_result

, the system first waits for the launch_done of the current batch, and then waits for the copy_done of the previous batch. Actually, the copy_done of the previous batch may complete earlier, but we cannot simply adjust the copy_done condition to trigger earlier and immediately proceed with the subsequent output processing, as it may cause CPU contention with the forward process. However, **is it possible to move the output processing after copy_done to a separate process? This would help avoid CPU contention with the forward process.**

We analyzed the waiting time difference between launch_done and copy_done. When launch_done occurs first, it takes an average of 9.58us for the copy_done condition to be satisfied. When copy_done occurs first, it takes an average of 14ms for the launch_done condition to be met, indicating that more time is spent on waiting for the output processing of the previous batch.

>     def resolve_last_batch_result(self, launch_done: Optional[threading.Event] = None):
        """
        This function is called to resolve the last batch result and
        wait for the current batch to be launched. Used in overlap mode.
        """
        copy_done, logits_output, next_token_ids, can_run_cuda_graph = (
            self.output_queue.get()
        )

        if launch_done is not None:
            launch_done.wait()
        copy_done.synchronize()

        if logits_output.next_token_logprobs is not None:
            logits_output.next_token_logprobs = (
                logits_output.next_token_logprobs.tolist()
            )
        if logits_output.input_token_logprobs is not None:
            logits_output.input_token_logprobs = tuple(
                logits_output.input_token_logprobs.tolist()
            )
        next_token_ids = next_token_ids.tolist()
        return logits_output, next_token_ids, can_run_cuda_graph

### Test Data

Server command:
> python3 -m sglang.launch_server --model-path  Qwen3-235B-A22B-Instruct-2507-FP8 --enable-metrics \
        --disable-radix-cache \
        --mem-fraction-static 0.65 \
        --tp-size 8 \
        --ep-size 8 \
        --enable-dp-attention --dp-size 8 \
        --chunked-prefill-size -1 \
        --enable-deepep-moe \
        --cuda-graph-max-bs 128 \
        --trust-remote-code \
        --attention-backend flashinfer \
        --disable-overlap-schedule

Test command:
> python -m sglang.bench_serving --backend sglang --dataset-name random --dataset-path /mnt/ShareGPT_V3_unfiltered_cleaned_split.json --random-input-len 3500 --random-output-len 10 --max-concurrency 240 --num-prompts 20 --request-rate 3.2 --random-range-ratio 1

Data with overlap scheduler enabled

> ============ Serving Benchmark Result ============
Backend:                                 sglang
Traffic request rate:                    3.2
Max request concurrency:                 240
Successful requests:                     20
Benchmark duration (s):                  5.28
Total input tokens:                      70000
Total generated tokens:                  200
Total generated tokens (retokenized):    200
Request throughput (req/s):              3.79
Input token throughput (tok/s):          13249.13
Output token throughput (tok/s):         37.85
Total token throughput (tok/s):          13286.98
Concurrency:                             13.89
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   3669.34
Median E2E Latency (ms):                 3999.02
---------------Time to First Token----------------
Mean TTFT (ms):                          2464.08
Median TTFT (ms):                        2422.23
P99 TTFT (ms):                           3227.57
---------------Inter-Token Latency----------------
Mean ITL (ms):                           133.92
Median ITL (ms):                         23.78
P95 ITL (ms):                            1076.42
P99 ITL (ms):                            2076.23
Max ITL (ms):                            2076.26
==================================================

Data with overlap scheduler disabled

> ============ Serving Benchmark Result ============
Backend:                                 sglang
Traffic request rate:                    3.2
Max request concurrency:                 240
Successful requests:                     20
Benchmark duration (s):                  5.34
Total input tokens:                      70000
Total generated tokens:                  200
Total generated tokens (retokenized):    200
Request throughput (req/s):              3.75
Input token throughput (tok/s):          13116.05
Output token throughput (tok/s):         37.47
Total token throughput (tok/s):          13153.52
Concurrency:                             14.03
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   3743.33
Median E2E Latency (ms):                 4071.25
---------------Time to First Token----------------
Mean TTFT (ms):                          1676.25
Median TTFT (ms):                        1674.84
P99 TTFT (ms):                           2236.25
---------------Inter-Token Latency----------------
Mean ITL (ms):                           229.67
Median ITL (ms):                         25.15
P95 ITL (ms):                            1141.01
P99 ITL (ms):                            2262.51
Max ITL (ms):                            2262.65
==================================================
